src/workerd/jsg/filesystem.c++

```diff
@@ -0,0 +1,537 @@
+          }
+          if (!fn(js, entry.key, dir->stat())) {
+            break;
+          }
+        }
+      }
+    }
+  }
+
+  kj::Maybe<kj::OneOf<kj::Own<File>, kj::Own<Directory>>> tryOpen(
+      Lock& js, kj::PathPtr path, kj::Maybe<Type> createAs) override {
+    if (path.size() == 0) {
+      return kj::Maybe<kj::OneOf<kj::Own<File>, kj::Own<Directory>>>(addRef(js));
+    }
+    KJ_IF_SOME(found, entries.find(path[0])) {
+      path = path.slice(1, path.size());
+      if (path.size() == 0) {
+        // We found the entry, return it.
+        KJ_SWITCH_ONEOF(found) {
+          KJ_CASE_ONEOF(file, kj::Own<File>) {
+            return kj::Maybe<kj::OneOf<kj::Own<File>, kj::Own<Directory>>>(file->addRef(js));
+          }
+          KJ_CASE_ONEOF(dir, kj::Own<Directory>) {
+            return kj::Maybe<kj::OneOf<kj::Own<File>, kj::Own<Directory>>>(dir->addRef(js));
+          }
+        }
+        KJ_UNREACHABLE;
+      }
+      KJ_SWITCH_ONEOF(found) {
+        KJ_CASE_ONEOF(file, kj::Own<File>) {
+          // We found a file, but we were looking for a directory.
+          return kj::none;
+        }
+        KJ_CASE_ONEOF(dir, kj::Own<Directory>) {
+          // We found a directory, continue searching.
+          return dir->tryOpen(js, path, createAs);
+        }
+      }
+    }
+
+    if constexpr (Writable) {
+      KJ_IF_SOME(type, createAs) {
+        return tryCreate(js, path, type);
+      }
+    } else {
+      JSG_REQUIRE(createAs == kj::none, Error,
+          "Cannot create a new file or directory in a read-only directory");
+    }
+
+    return kj::none;
+  }
+
+  void setLastModified(Lock& js, kj::Date date = kj::UNIX_EPOCH) override {
+    if constexpr (Writable) {
+      lastModified = date;
+    }
+  }
+
+  kj::Own<File> transfer(Lock& js, kj::StringPtr name, kj::Own<File>&& file) override {
+    if constexpr (Writable) {
+      validatePathWithNoSeparators(name);
+      KJ_IF_SOME(_, entries.find(name)) {
+        JSG_FAIL_REQUIRE(Error, "File or directory already exists: \"", name, "\"");
+      }
+      auto ret = file->addRef(js);
+      entries.upsert(
+          kj::str(name), kj::mv(file), [&](auto& current, auto&& node) { return kj::mv(node); });
+      return kj::mv(ret);
+    } else {
+      JSG_FAIL_REQUIRE(Error, "Cannot transfer a file into a read-only directory");
+    }
+  }
+
+  kj::Own<Directory> transfer(Lock& js, kj::StringPtr name, kj::Own<Directory>&& dir) override {
+    if constexpr (Writable) {
+      validatePathWithNoSeparators(name);
+      KJ_IF_SOME(_, entries.find(name)) {
+        JSG_FAIL_REQUIRE(Error, "File or directory already exists: \"", name, "\"");
+      }
+      auto ret = dir->addRef(js);
+      entries.upsert(
+          kj::str(name), kj::mv(dir), [&](auto& current, auto&& node) { return kj::mv(node); });
+      return kj::mv(ret);
+    } else {
+      JSG_FAIL_REQUIRE(Error, "Cannot transfer a directory into a read-only directory");
```

Comment:
I feel like this should be the first line in this function

Reply:
Not following... this is in a `if consexpr(...)` else clause. It's only relevant to that branch.